bigtable/src/main/scala/magnolify/bigtable/BigtableType.scala

```diff
@@ -164,22 +162,82 @@ object BigtableField {
     override def fromByteString(v: ByteString): ByteString = v
     override def toByteString(v: ByteString): ByteString = v
   }
-  implicit val btfByteArray = from[ByteString](_.toByteArray)(ByteString.copyFrom(_))
-  implicit val btfString = from[ByteString](_.toStringUtf8)(ByteString.copyFromUtf8(_))
+  implicit val btfByteArray = from[ByteString](_.toByteArray)(ByteString.copyFrom)
+  implicit val btfString = from[ByteString](_.toStringUtf8)(ByteString.copyFromUtf8)
 
   implicit def btfOption[A](implicit btf: BigtableField[A]): BigtableField[Option[A]] =
     new BigtableField[Option[A]] {
-      override def get(xs: java.util.List[RowCell], k: String): Option[A] = {
-        val isOption = xs.asScala.exists { c =>
-          val q = c.getQualifier.toStringUtf8
-          q == k || q.startsWith(s"$k.") // optional primitive or nested field
+      override def get(xs: java.util.List[RowCell], k: String): Option[A] =
+        RowCells.findNullable(xs, k).map(btf.get(_, k))
+
+      override def put(k: String, v: Option[A]): Seq[SetCell.Builder] =
+        v.toSeq.flatMap(btf.put(k, _))
+    }
+}
+
+private object RowCells {
+  private val cmp = ByteStringComparator.INSTANCE
+
+  private def find(
```

Comment:
Just wanted to mention that searching a cell in a list of RowCells wouldn't be that efficient as looking up a rowCell directly in a row using Row.getCells(), but looks like you already solved this issue :)